File: src/main/java/com/ssoon/WorkingWith2DArrays.java
package com.ssoon;

import java.util.Arrays;

public class WorkingWith2DArrays {

    public static void main(String[] args) {
        var board = new char[3][3];
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                board[i][j] = '-';
            }
        }

        var boardTwo = new char[][]{
                new char[]{'O', '-', '-'},
                new char[]{'O', '-', '-'},
                new char[]{'O', '-', '-'}
        };

        board[0][0] = 'O';
        board[1][0] = 'O';
        board[2][0] = 'O';

        System.out.println("Arrays.deepToString(board) = " + Arrays.deepToString(board));
        System.out.println("Arrays.deepToString(boardTwo) = " + Arrays.deepToString(boardTwo));
    }
}

File: src/main/kotlin/com/ssoon/WorkingWith2DArrays.kt
package com.ssoon

fun main() {
    val board = Array(3) { CharArray(3) }
    for (i in 0..2) {
        for (j in 0..2) {
            board[i][j] = '-'
        }
    }

    board[0][0] = 'O'
    board[1][0] = 'O'
    board[2][0] = 'O'

    val boardTwo = arrayOf(
        charArrayOf('O', '-', '-'),
        charArrayOf('O', '-', '-'),
        charArrayOf('O', '-', '-')
    )

    println("Arrays.deepToString(board) = " + board.contentDeepToString())
    println("Arrays.deepToString(board) = " + boardTwo.contentDeepToString())
}